### PR TITLE
LPD-11274 Support for SugarCRM connector for Proxy Objects

### PR DESCRIPTION
**Pull request by guilhermedcamacho**

### CODE REVIEW

File: modules/apps/object/object-storage-sugarcrm/src/main/java/com/liferay/object/storage/sugarcrm/internal/rest/manager/v1_0/SugarCRMObjectEntryManagerImpl.java [hunk 1]
```diff
@@ -0,0 +1,570 @@
...
+				dtoConverterContext, objectDefinition, objectEntry,
+				_getTriConsumer(objectDefinition)));
+
+		return toObjectEntry(
+			objectDefinition.getCompanyId(), _getDateFormat(),
+			_defaultObjectFieldNames, dtoConverterContext, responseJSONObject,
+			objectDefinition);
+	}
+
+	private void _appendFilter(StringBuilder sb, String filterString) {
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-storage-sugarcrm/src/main/java/com/liferay/object/storage/sugarcrm/internal/rest/manager/v1_0/SugarCRMObjectEntryManagerImpl.java:197]: Hey @mauricesepe , you implemented the filter feature following the SugarCRM standard, but this should work following the Liferay (Odata) standard and then translating it to the SugarCRM standard. Please follow the pattern used in Salesforce implementation (see classes [SalesforceFilterFactoryImpl](https://github.com/liferay/liferay-portal/blob/master/modules/apps/object/object-storage-salesforce/src/main/java/com/liferay/object/storage/salesforce/internal/filter/factory/SalesforceFilterFactoryImpl.java) and [SOSQLExpressionVisitorImpl](https://github.com/liferay/liferay-portal/blob/master/modules/apps/object/object-storage-salesforce/src/main/java/com/liferay/object/storage/salesforce/internal/odata/filter/expression/SOSQLExpressionVisitorImpl.java)). Thanks!

**Comment by guilhermedcamacho** [comment]
ci:test:sf

**Comment by brianchandotcom** [comment]
@guilhermedcamacho ETA to the PR getting to me?

**Comment by guilhermedcamacho** [comment]
> @guilhermedcamacho ETA to the PR getting to me?

I will send it to you by the end of this week.

**Comment by guilhermedcamacho** [comment]
ci:test:sf

**Comment by guilhermedcamacho** [comment]
ci:test:sf

**Comment by guilhermedcamacho** [comment]
ci:test:sf

**Comment by guilhermedcamacho** [comment]
ci:test:relevant

**Comment by guilhermedcamacho** [comment]
ci:test:object

**Comment by guilhermedcamacho** [comment]
Manually forwarded to [brianchandotcom#148714](https://github.com/brianchandotcom/liferay-portal/pull/148714)

**Pull request closed without merging**